src/frontend/src/table/insert.rs

```diff
@@ -71,8 +71,8 @@ impl DistTable {
         for join in joins {
             let object_result = join.await.context(error::JoinTaskSnafu)??;
             let result = match object_result {
-                client::ObjectResult::Select(_) => unreachable!(),
                 client::ObjectResult::Mutate(result) => result,
+                _ => unreachable!(),
```

Comment:
I'd prefer enumerate all variants instead of wildcard match arm, in that when adding a new variant to `ObjectResult`, enumeration will raise a compile error while wildcard arm silently swallow the new variant.